aea/protocols/dialogue/base.py

```diff
@@ -1309,6 +1344,121 @@ def teardown(self) -> None:
+        dialogue = super().get(dialogue_label)
+        if dialogue:
+            return dialogue
+
+        dialogue = self._get_dialogue_from_collection(
+            dialogue_label, self._terminal_dialogues_collection
+        )
+        if dialogue:
+            # get dialogue from terminal state collection and cache it
+            self._add_terminal_state_dialogue(dialogue)
+            return dialogue
+        return None
+
+    def _get_dialogue_from_collection(
+        self, dialogue_label: "DialogueLabel", collection: SyncCollection
+    ) -> Optional[Dialogue]:
+        """
+        Get dialogue by label from collection.
+
+        :param dialogue_label: label for lookup
+        :param collection: collection with dialogues
+        :return: dialogue if exists
+        """
+        if not collection:
+            return None
+        dialogue_data = collection.get(str(dialogue_label))
+        if not dialogue_data:
+            return None
+        dialogue_data = cast(Dict, dialogue_data)
+        return self._dialogue_from_json(dialogue_data)
+
+    def _load_terminated_dialogues(self) -> None:
+        """Skip terminated dialogues loading, cause it's offloaded."""
+
+    def _get_dialogues_by_address_from_collection(
+        self, address: Address, collection: SyncCollection
+    ) -> List["Dialogue"]:
+        """
+        Get all dialogues with opponent address from specified collection.
+
+        :param address: address for lookup.
+        :param: collection: collection to get dialogues from.
+
+        :return: list of dialogues
+        """
+        if not collection:
+            return []
+
+        return [
+            self._dialogue_from_json(cast(Dict, i[1]))
+            for i in collection.find("dialogue_label.dialogue_opponent_addr", address)
+        ]
+
+    def get_dialogues_with_counterparty(self, counterparty: Address) -> List[Dialogue]:
+        """
+        Get the dialogues by address.
+
+        :param counterparty: the counterparty
+        :return: The dialogues with the counterparty.
+        """
+        return list(
+            {
+                i.dialogue_label: i
+                for i in (
+                    super().get_dialogues_with_counterparty(counterparty)
+                    + self._get_dialogues_by_address_from_collection(
+                        counterparty, self._active_dialogues_collection
+                    )
+                    + self._get_dialogues_by_address_from_collection(
+                        counterparty, self._terminal_dialogues_collection
+                    )
+                )
+            }.values()
+        )
+
+    @property
+    def dialogues_in_terminal_state(self) -> List["Dialogue"]:
+        """Get all dialogues in terminal state."""
+        dialogues = super().dialogues_in_terminal_state + list(
+            self._load_dialogues(self._terminal_dialogues_collection)
+        )
+        return list({i.dialogue_label: i for i in dialogues}.values())
```

Comment:
don't use single char vars; just call it `dialogue`

Reply:
Also, what's the point of this line?

Reply:
discussed